drivers/adc/ad469x/ad469x.c

```diff
@@ -0,0 +1,524 @@
+
+	ret = ad469x_spi_write_mask(dev,
+				    AD469x_REG_GP_MODE,
+				    AD469x_GP_MODE_BUSY_GP_EN_MASK,
+				    AD469x_GP_MODE_BUSY_GP_EN(1));
+
+	ret |= ad469x_spi_write_mask(dev,
+				     AD469x_REG_GP_MODE,
+				     AD469x_GP_MODE_BUSY_GP_SEL_MASK,
+				     AD469x_GP_MODE_BUSY_GP_SEL(gp_sel));
+
+	return ret;
+}
+
+/**
+ * @brief Enter conversion mode.
+ *        To exit conversion mode send a 5 bit conversion mode command
+ *        AD469x_CMD_REG_CONFIG_MODE
+ * @param [in] dev - ad469x_dev device handler.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad469x_enter_conversion_mode(struct ad469x_dev *dev)
+{
+	return ad469x_spi_write_mask(dev,
+				     AD469x_REG_SETUP,
+				     AD469x_SETUP_IF_MODE_MASK,
+				     AD469x_SETUP_IF_MODE_CONV);
+}
+
+/**
+ * @brief Exit conversion mode.
+ *        Enter register mode to read/write registers
+ * @param [in] dev - ad469x_dev device handler.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad469x_exit_conversion_mode(struct ad469x_dev *dev)
+{
+	uint32_t commands_data[1], buf;
+	struct spi_engine_offload_message msg;
+	uint32_t spi_eng_msg_cmds[3] = {
+		CS_LOW,
+		WRITE_READ(1),
+		CS_HIGH
+	};
+	int32_t ret;
+
+	pwm_enable(dev->trigger_pwm_desc);
+
+	commands_data[0] = AD469x_CMD_REG_CONFIG_MODE << 8;
+
+	ret = spi_engine_offload_init(dev->spi_desc, dev->offload_init_param);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	msg.commands = spi_eng_msg_cmds;
+	msg.no_commands = ARRAY_SIZE(spi_eng_msg_cmds);
+	msg.rx_addr = (uint32_t)&buf;
+	msg.commands_data = commands_data;
+
+	ret = spi_engine_offload_transfer(dev->spi_desc, msg, 1);
+	if (ret != SUCCESS)
+		return ret;
+
+	pwm_disable(dev->trigger_pwm_desc);
+	if (ret != SUCCESS)
+		return ret;
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Read from device.
+ *        Enter register mode to read/write registers
+ * @param [in] dev - ad469x_dev device handler.
+ * @param [in] channel - ad469x selected channel.
+ * @param [out] buf - data buffer.
+ * @param [in] samples - sample number.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad469x_read_data(struct ad469x_dev *dev,
+			 uint8_t channel,
+			 uint32_t *buf,
+			 uint16_t samples)
+{
+	int32_t ret;
+	uint32_t commands_data[1];
+	struct spi_engine_offload_message msg;
+	uint32_t spi_eng_msg_cmds[3] = {
+		CS_LOW,
+		WRITE_READ(1),
+		CS_HIGH
+	};
+
+	commands_data[0] = AD469x_CMD_CONFIG_CH_SEL(channel) << 8; /* ch 0 */
+
+	pwm_enable(dev->trigger_pwm_desc);
+
+	ret = spi_engine_offload_init(dev->spi_desc, dev->offload_init_param);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	msg.commands = spi_eng_msg_cmds;
+	msg.no_commands = ARRAY_SIZE(spi_eng_msg_cmds);
+	msg.rx_addr = (uint32_t)buf;
+	msg.commands_data = commands_data;
+
+	ret = spi_engine_offload_transfer(dev->spi_desc, msg, samples * 2);
+	if (ret != SUCCESS)
+		return ret;
+
+	Xil_DCacheInvalidateRange(msg.rx_addr, samples * 4);
```

Comment:
why is there a call to a specific xilinx function ? The driver should be platform agnostic.